From 004f30ed7c2c65aa14e1bd35b746e8bcde37d6ae Mon Sep 17 00:00:00 2001
From: Robert Lehmann <robert.lehmann@sitec-systems.de>
Date: Mon, 22 Feb 2016 16:05:03 +0100
Subject: [PATCH] S2: Remove init of usdhc3

The initialisation and pin muxing of usdhc3 is removed. This
initialisation causes a performance leak in the can communication
because of the broken-cd option.

Ticket #954
---
 arch/arm/boot/dts/imx6sx-s2.dts | 73 -----------------------------------------
 1 file changed, 73 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx-s2.dts b/arch/arm/boot/dts/imx6sx-s2.dts
index 72698c2..b73e800 100644
--- a/arch/arm/boot/dts/imx6sx-s2.dts
+++ b/arch/arm/boot/dts/imx6sx-s2.dts
@@ -36,24 +36,6 @@
             gpios = <&gpio3 20 GPIO_ACTIVE_HIGH>;
         };
     };
-
-	/*regulators {*/
-		/*compatible = "simple-bus";*/
-		/*#address-cells = <1>;*/
-		/*#size-cells = <0>;*/
-
-		/*reg_5p0: regulator@4 {*/
-			/*compatible = "regulator-fixed";*/
-			/*reg = <0>;*/
-			/*pinctrl-names = "default";*/
-			/*pinctrl-0 = <&pinctrl_5pe>;*/
-			/*regulator-name = "5p0";*/
-			/*regulator-min-microvolt = <5000000>;*/
-			/*regulator-max-microvolt = <5000000>;*/
-			/*gpio = <&gpio2 12 GPIO_ACTIVE_LOW>;*/
-			/*regulator-always-on;*/
-		/*};*/
-	/*};*/
 };
 
 &cpu0 {
@@ -123,25 +105,6 @@
 	status = "okay";
 };
 
-/*&usdhc2 {*/
-	/*pinctrl-names = "default";*/
-	/*pinctrl-0 = <&pinctrl_usdhc2>;*/
-	/*bus-width = <4>;*/
-	/*non-removable;*/
-	/*no-1-8-v;*/
-	/*keep-power-in-suspend;*/
-	/*enable-sdio-wakeup;*/
-	/*status = "okay";*/
-/*};*/
-
-&usdhc3 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_usdhc3>;
-	bus-width = <4>;
-    broken-cd;
-	status = "okay";
-};
-
 &i2c2 {
 	clock-frequency = <100000>;
 	pinctrl-names = "default";
@@ -217,12 +180,6 @@
 	pinctrl-0 = <&pinctrl_hog>;
 
 	imx6x-score {
-		/*pinctrl_5pe: power_5P0_enable {*/
-			/*fsl,pins = <*/
-				/*MX6SX_PAD_KEY_COL2__GPIO2_IO_12		0x80*/
-			/*>;*/
-		/*};*/
-
         pinctrl_hog: hoggrp {
 			fsl,pins = <
 				MX6SX_PAD_LCD1_HSYNC__GPIO3_IO_26	0x17059
@@ -272,14 +229,6 @@
 			>;
 		};
 
-		pinctrl_can_gpios: can-gpios {
-			fsl,pins = <
-				/* CAN_STANDBY */
-				/*MX6SX_PAD_GPIO1_IO02__GPIO1_IO_2	0x17059*/
-				MX6SX_PAD_GPIO1_IO02__GPIO1_IO_2	0xc9
-			>;
-		};
-
 		pinctrl_ecspi3_cs: ecspi3_cs_grp {
 			fsl,pins = <
 				MX6SX_PAD_SD4_DATA7__ECSPI3_SS0		0x80000000
@@ -329,27 +278,5 @@
 				MX6SX_PAD_KEY_ROW1__UART6_RX		0x1b0b1
 			>;
 		};
-
-		/*pinctrl_usdhc2: usdhc2grp {*/
-			/*fsl,pins = <*/
-				/*MX6SX_PAD_SD2_CMD__USDHC2_CMD		0x17059*/
-				/*MX6SX_PAD_SD2_CLK__USDHC2_CLK		0x10059*/
-				/*MX6SX_PAD_SD2_DATA0__USDHC2_DATA0	0x17059*/
-				/*MX6SX_PAD_SD2_DATA1__USDHC2_DATA1	0x17059*/
-				/*MX6SX_PAD_SD2_DATA2__USDHC2_DATA2	0x17059*/
-				/*MX6SX_PAD_SD2_DATA3__USDHC2_DATA3	0x17059*/
-			/*>;*/
-		/*};*/
-
-		pinctrl_usdhc3: usdhc3grp {
-			fsl,pins = <
-				MX6SX_PAD_SD3_CMD__USDHC3_CMD		0x17069
-				MX6SX_PAD_SD3_CLK__USDHC3_CLK		0x10071
-				MX6SX_PAD_SD3_DATA0__USDHC3_DATA0	0x17069
-				MX6SX_PAD_SD3_DATA1__USDHC3_DATA1	0x17069
-				MX6SX_PAD_SD3_DATA2__USDHC3_DATA2	0x17069
-				MX6SX_PAD_SD3_DATA3__USDHC3_DATA3	0x17069
-			>;
-		};
 	};
 };
-- 
1.9.1

